lib/project_types/script/layers/infrastructure/languages/assemblyscript_project_creator.rb

```diff
@@ -11,23 +11,73 @@ class AssemblyScriptProjectCreator
           property! :script_name, accepts: String
           property! :path_to_project, accepts: String
 
-          BOOTSTRAP = "npx --no-install shopify-scripts-toolchain-as bootstrap --from %{extension_point} --dest %{base}"
           BUILD = "shopify-scripts-toolchain-as build --src src/shopify_main.ts " \
           "--binary build/script.wasm --metadata build/metadata.json"
-          MIN_NODE_VERSION = "14.5.0"
+          MIN_NODE_VERSION = "14.5.0" # kept because task_runner uses this
           ASC_ARGS = "-- --lib node_modules --optimize --use Date="
 
+          EP_REPO = "https://github.com/Shopify/extension-points.git"
+          ORIGIN_BRANCH = 'master'
+
           def setup_dependencies
+            # sparse checkout actions
+            git_init
+            setup_remote
+            setup_sparse_checkout
+            pull
+            clean
+
+            # TODO: files that should be in the repo, but arent yet
             write_npmrc
             write_package_json
           end
 
           def bootstrap
-            command_runner.call(bootstap_command)
           end
 
           private
 
+          def git_init
```

Comment:
What do you think about moving these into the existing [git class](https://github.com/Shopify/shopify-app-cli/blob/master/lib/shopify-cli/git.rb)?